### PR TITLE
Swarm comm debug fix

### PR DESCRIPTION
**Pull request by brryan**
## PR Summary

@AstroBarker noticed his downstream tracers code was crashing in Debug mode. Turns out the particles tracers example also crashes in Debug mode. This was due to accessing neighbor buffer indices even for particles that shouldn't be getting sent anywhere because they're currently in the active meshblock (and so their neighbor buffer index is `-1`). Simple fix.

## PR Checklist

- [x] Code passes cpplint
- [ ] New features are documented.
- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [ ] Change is breaking (API, behavior, ...)
  - [ ] Change is *additionally* added to CHANGELOG.md in the breaking section
  - [ ] PR is marked as breaking
  - [ ] Short summary API changes at the top of the PR (plus optionally with an automated update/fix script)
- [ ] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [x] Docs build
- [x] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

File: src/interface/swarm_comms.cpp [hunk 1]
```diff
@@ -250,18 +250,20 @@ void Swarm::LoadBuffers_() {
...
-              swarm_d.MarkParticleForRemoval(p_index);
-              for (int i = 0; i < realPackDim; i++) {
-                bdvar.send[bufid](buffer_index) = vreal(i, p_index);
-                buffer_index++;
-              }
-              for (int i = 0; i < intPackDim; i++) {
-                bdvar.send[bufid](buffer_index) = static_cast<Real>(vint(i, p_index));
-                buffer_index++;
+              const int bufid = neighbor_buffer_index(m);
+              if (m >= 0) {
```
Comment (Yurlungur) [inline src/interface/swarm_comms.cpp]: Is this `m >=0` duplicated with line 253 above? Is that intentional?

**Review by Yurlungur — APPROVED** [review]
Thanks for the quick fix! This was a blocker in Phoebus.

File: src/interface/swarm_comms.cpp (hunk 1 above)
Comment (brryan) [reply src/interface/swarm_comms.cpp]: Thanks @Yurlungur, not sure how I missed that, fixed. :+1:

**Comment by AstroBarker** [comment]
Thanks for the quick find! This seems to have done the trick.

**Review by AstroBarker — APPROVED** [review]

**Pull request merged**